### DISCUSSION TITLE
Can you pickle a fitted knn regressor?

### DISCUSSION THREAD

**Discussion by ecavan** [GAA-UAM/scikit-fda#460] [Q&A] [answered]
Having trouble saving the trained mode, "Can't pickle <function _to_multivariate_metric.<locals>.multivariate_metric
![error](https://user-images.githubusercontent.com/12787188/178523632-4b39b564-c96e-47be-b3ec-d8801848af10.PNG)
"

**Comment by vnmabus** [comment]
Which version of the software are you using?

Can you provide a minimal example?

**Reply by ecavan** [reply]
Python: 3.9.12
scikit-fda==0.7.1
sklearn == 1.0.2

I was able to save the model by using the multivariate_metric=True parameter with a small decrease to the testing score when I reloaded it

**Reply by vnmabus** [reply, marked as answer]
The PR ##465, merged to the develop branch should fix this, as well as improving performance significantly for large datasets. Please test the develop version to see if it fixes your problem.
